public-images/public-images-how-to/launch-with-libvirt.rst

```diff
@@ -0,0 +1,147 @@
+--------------------------
+
+Ubuntu cloud images are hosted on https://cloud-images.ubuntu.com/
+
+QCOW2 images (``.img``) are suitable for use with `libvirt`_ and the `QEMU`_ driver.
+Refer to :doc:`../public-images-reference/artifacts` for a description the various image types.
+
+Once you have identified a suitable image, download it. For example, the following would download the current daily Ubuntu 24.04 (noble) image for amd64 machines:
+
+.. code:: bash
+
+   wget https://cloud-images.ubuntu.com/noble/current/noble-server-cloudimg-amd64.img
+
+
+
+Launching through console
+-------------------------
+
+The following command will
+
+* Create a new VM named ``ubuntu-vm-name``.
+* Skip attaching to the console.
+* Imports the image (as opposed to running an installer from it).
+* Create a machine with 2GB of RAM and 2 virtual CPUs.
+* Select a generic OS. Although it might be tempting to specify Ubuntu for ``--osinfo``, this would have the effect of trying to automate an installation. Cloud images are pre-installed.
+* Select the virtio disk driver and the path to a downloaded disk image. This path needs to be accessible by the ``libvirt-daemon``.
+* Attach the VM to the ``default`` network.
+* Specify the cloud-init user-data file previously created.
+
+
+.. code:: bash
+
+   virt-install \
+      --name ubuntu-vm-name \
+      --noautoconsole \
+      --import \
+      --memory 2048 --vcpus=2 \
+      --osinfo generic \
+      --disk bus=virtio,path=noble-server-cloudimg-amd64.img \
+      --network default \
+      --cloud-init user-data=user-data
+
+
+Then, you can then interact with the newly created virtual machine with ``virsh``, ``virt-manager`` or any other libvirt application.
+
+To attach to the console
+
+.. code:: bash
+
+   virsh console ubuntu-vm-name
+
+Once attached, you can also detach from the console by pressing ``Ctrl-]``
+
+
+To shutdown the VM
+
+.. code:: bash
+
+   virsh shutdown ubuntu-vm-name
+
+
+To remove the VM and its disk image
+
+.. code:: bash
+
+   virsh undefine --domain ubuntu-vm-name --remove-all-storage
+
+
+
+Alternatively using ``uvtool-libvirt``
+--------------------------------------
+
+Finding, downloading and launching images can be simplified by the use of uvtool. uvtool can look at published cloud images *simplestreams*, retrieve and launch VM images through `libvirt`.
+
+.. code:: bash
+
+   uvt-simplestreams-libvirt sync --source STREAM_URL FILTERS
+
+Common stream URLs are
+
+* releases: https://cloud-images.ubuntu.com/releases/
+* daily: https://cloud-images.ubuntu.com/releases/
+* minimal releases: https://cloud-images.ubuntu.com/minimal/releases/
+* minimal daily: https://cloud-images.ubuntu.com/minimal/daily/
+
+*FILTERS* are a combination of image attributes. If multiple images the combination of filters they will all be downloaded. For example, the following would download the most recent images matching both the amd64 architecture AND the Noble Numbat (24.04) Ubuntu release.
```

Comment:
```suggestion
*FILTERS* are a combination of image attributes. If multiple images match the combination of filters they will all be downloaded. For example, the following would download the most recent images matching both the amd64 architecture AND the Noble Numbat (24.04) Ubuntu release.
```

Reply:
Also is there a doc which describes the full set of possible filters? and is `ARCH` meant to be all caps in the following command?

Reply:
I've added a short description of the fields present on our streams, but it's not the full picture and there is a jira card to cover simplestreams